### PR TITLE
Make progress on the parsers transition to concrete syntax

### PR DESCRIPTION
**Pull request by pmderodat**

### CODE REVIEW

File: testsuite/tests/grammar/indent_trivia/expected_concrete_syntax.lkt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-grammar None_grammar {
+grammar FooGrammar {
```
Comment (raph-amiard) [inline testsuite/tests/grammar/indent_trivia/expected_concrete_syntax.lkt]: foo_grammar would be better if we keep capitalized names for types

File: langkit/parsers.py [hunk 2]
```diff
@@ -382,6 +398,175 @@ def check_main_rule(self, context):
                     )
                 )
 
+    @classmethod
+    def create_from_lkt(cls, ctx, lkt_unit):
```
Comment (raph-amiard) [inline langkit/parsers.py]: As discussed live, moving the new compilation logic to a separate file would seem better to me

File: testsuite/tests/grammar/indent_trivia/expected_concrete_syntax.lkt (hunk 1 above)
Comment (pmderodat) [reply testsuite/tests/grammar/indent_trivia/expected_concrete_syntax.lkt]: Will fix.

File: langkit/compile_context.py [hunk 3]
```diff
@@ -689,6 +702,24 @@ def __init__(self, lang_name, lexer, grammar,
...
+
+        :param str lkt_file: Name of the file to parse.
+        :rtype: liblktlang.AnalysisUnit
+        """
+        import liblktlang
+
+        basename = os.path.basename(lkt_file)
+        result = liblktlang.AnalysisContext().get_from_file(lkt_file)
+        if result.diagnostics:
+            for d in result.diagnostics:
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply langkit/compile_context.py]: Agreed. As discussed live: let’s put a TODO and do this only once the transition is done: we’ll be able to simplify the diagnostics mechanism at this point, making it easier to reuse it for parsing errors.

File: langkit/parsers.py [hunk 4]
```diff
@@ -382,6 +398,175 @@ def check_main_rule(self, context):
...
+        """
+        Load a grammar from a Lktlang source file.
+
+        :param liblktlang.AnalysisUnit lkt_unit: Analysis unit where to look
+            for the grammar.
+        :rtype: langkit.parsers.Grammar
+        """
+        import liblktlang
+
+        def annotations(decl):
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply langkit/parsers.py]: Sure: will do as another step as discussed live.

File: langkit/parsers.py (hunk 2 above)
Comment (pmderodat) [reply langkit/parsers.py]: Sure: will do as another step as discussed live.

**Pull request merged**